### PR TITLE
add credential validation

### LINKED ISSUE #8972: Add validation for GCP, Azure, AWS, DO

### PR DESCRIPTION
**Pull request by zreigz**
**What does this PR do / Why do we need it**: add credential validation for DO, GCP, Azure and AWS

**Does this PR close any issues?**:
Fixes #8972

```release-note
NONE
```

### CODE REVIEW

File: pkg/provider/cloud/azure/client.go [hunk 1]
```diff
@@ -158,6 +160,18 @@ func GetCredentialsForCluster(cloud kubermaticv1.CloudSpec, secretKeySelector pr
 	}, nil
 }
 
+func ValidateCredentials(credentials Credentials) error {
```
Comment (xrstf) [inline pkg/provider/cloud/azure/client.go]: Can we please have this as `ValidateCredentials(ctx context.Context, credentials Credentials) error` for all providers?

**Review by xrstf — CHANGES_REQUESTED** [review]
Please integrate contexts.

File: pkg/provider/cloud/azure/client.go (hunk 1 above)
Comment (zreigz) [reply pkg/provider/cloud/azure/client.go]: done

File: pkg/provider/cloud/digitalocean/provider.go [hunk 2]
```diff
@@ -45,17 +45,21 @@ func (do *digitalocean) DefaultCloudSpec(spec *kubermaticv1.CloudSpec) error {
 	return nil
 }
 
+func ValidateCredentials(ctx context.Context, token string) error {
+	static := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	client := godo.NewClient(oauth2.NewClient(context.Background(), static))
```
Comment (xrstf) [inline pkg/provider/cloud/digitalocean/provider.go]: Leftover background context

**Review by xrstf — COMMENTED** [review]
One last minor leftover, then ❤️

File: pkg/provider/cloud/digitalocean/provider.go (hunk 2 above)
Comment (zreigz) [reply pkg/provider/cloud/digitalocean/provider.go]: done

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**